pkg/crd-schema/yaml/nested.go

```diff
@@ -0,0 +1,63 @@
+package yaml
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+// GetNestedFieldNoCopy gets a value at a path in a YAML in-memory object.
+func GetNestedFieldNoCopy(x interface{}, pth ...string) (interface{}, bool, error) {
```

Comment:
All these belong to sig.k8s.io/yaml, with proper tests. Basically resemble to JSON based funcs in the apimachinery unstructured package, but operating on `yaml.MapSlice` based objects.

Reply:
why aren't you just using `unstructured.Unstructured`, which already has these, IIRC?

Reply:
JSON destroys order of existing files. We can't do that. Those files come from other sources (most are manually written).

Reply:
added tests